packages/@aws-cdk/cdk/lib/cloudformation/output.ts

```diff
@@ -124,6 +132,19 @@ export class Output extends StackElement {
   public get ref(): string {
     throw new Error('Outputs cannot be referenced');
   }
+
+  /**
+   * Automatically determine an output name for use with FnImportValue
+   *
+   * This gets called in case the user hasn't specified an export name but is
+   * taking an action that requires exporting. We namespace with the stack name
+   * to reduce chances of collissions between CDK apps.
+   */
+  private uniqueOutputName() {
+    // prefix export name with stack name since exports are global within account + region.
+    const stackName = require('./stack').Stack.find(this).name;
+    return (stackName ? stackName + ':' : '') + this.logicalId;
```

Comment:
Why bother for the case of an un-named stack? I guess this only ever happens in unit tests, where this shouldn't matter. I'd be up for using

```ts
return `${stackName}:${this.logicalId}`;
```